docs/canonicalk8s/snap/reference/architecture.md

```diff
@@ -12,61 +10,44 @@ Kubernetes with users and with other systems.
-  administrator who deployed the workload they are interested in.
-
-There are non-human users of the K8s snap, for example the [`k8s-operator
-charm`][K8s charm]. The K8s charm needs to drive the Kubernetes cluster and to
-orchestrate the multi-node clustering operations.
-
-A set of external systems need to be easily integrated with our K8s
-distribution. We have identified the following:
-
-- **Load Balancer**: Although the K8s snap distribution comes with a
-   load balancer we expect the end customer environment to have a load balancer
-   and thus we need to integrate with it.
-- **Storage**: Kubernetes typically expects storage to be external to the
-  cluster. The K8s snap comes with a local storage option but we still need to
-  offer proper integration with any storage solution.
-- **Identity management**: Out of the box the K8s snap offers credentials for
-  an admin user. The admin user can complete the integration with any identity
-  management system available or do user management manually.
-- **External datastore**: By default, Kubernetes uses etcd to keep track of
-  state. Our K8s snap comes with `dqlite` as its datastore. We should however
-  be able to use any end client owned datastore installation. That should
-  include an external `postgresql` or `etcd`.
+Actors that interact with the K8s snap:
+
+- **K8s admin**:  interacts directly with the Kubernetes API server. {{product}}
+provides out of the box admin access to configure the cluster.
+- **K8s user** : consumes the workloads hosted in the cluster.
+
+Non-human users of the K8s snap:
+
+- [`k8s-operator charm`][K8s charm]: uses the snap to drive the Kubernetes
+cluster and orchestrate multi-node clustering operations.
+
+Although {{product}} provides its own implementation of the following services,
+external systems can be easily integrated:
+
+- **Load Balancer**
+- **Storage**
+- **Identity management**
+- **External datastore**
 
 ## The k8s snap
 
-Looking more closely at what is contained within the K8s snap itself:
+Contained within the K8s snap itself:
 
 ![cluster1][]
 
 The `k8s` snap distribution includes the following:
 
 - **Kubectl**: through which users and other systems interact with Kubernetes
-  and drive the cluster operations.
-- **K8s services**: These are all the Kubernetes services as well as core
-  workloads built from upstream and shipped in the snap.
-- State is backed up by **dqlite** by default, which keeps that state of the
-  Kubernetes cluster as well as the state we maintain for the needs of the
-  cluster operations. The cluster state may optionally be stored in a
-  different, external datastore.
-- **Runtime**: `containerd` and `runc` are the shipped container runtimes.
-- **K8sd**: which implements the operations logic and exposes that
-  functionality via CLIs and APIs.
+and drive the cluster operations.
+- **K8s control plane**: the Kubernetes services, as well as core
+workloads built from upstream and shipped in the snap.
```

Comment:
In the upstream docs, they refer to them as [`core components`](https://kubernetes.io/docs/concepts/overview/components/#core-components). Maybe we could align our definitions with that?
```suggestion
- **K8s core components**: the Kubernetes services, as well as core
workloads built from upstream and shipped in the snap.
```

Reply:
Do you think we should also update the diagrams?

Reply:
Sorry @nhennigan! I missed the diagrams, they are hard to see in the Github pane. But yes, let's change that to K8s core components and we can reuse the same description from the list.